sky/spot/spot_state.py

```diff
@@ -388,104 +423,122 @@ def get_nonterminal_job_ids_by_name(name: Optional[str]) -> List[int]:
 
     name_filter = ''
     if name is not None:
-        name_filter = 'AND (spot.job_name=(?) OR job_names.name=(?))'
+        name_filter = 'AND (spot.job_name=(?) OR job_info.name=(?))'
         field_values.extend([name, name])
 
     statuses = ', '.join(['?'] * len(SpotStatus.terminal_statuses()))
     rows = _CURSOR.execute(
         f"""\
-        SELECT DISTINCT spot.new_job_id
+        SELECT DISTINCT spot.spot_job_id
         FROM spot
-        LEFT OUTER JOIN job_names
-        ON spot.new_job_id=job_names.new_job_id
+        LEFT OUTER JOIN job_info
```

Comment:
Document why we do LEFT OUTER JOIN? IIUC, for old jobs? (job_info has not been populated?)

Similarly, doc on L426 why we do an OR?

Reply:
Good point! Added.